src/hooks/useTokenIdentityRegistryAgents.ts

```diff
@@ -34,6 +37,33 @@ export function useTokenIdentityRegistryAgents(registry?: EvmAddress) {
     }
   }, [registry]);
 
+  const filteredNotAgentsYetQueryResult = useQueries({
+    queries: (identityItems ?? []).map((agent) => ({
+      queryKey: [QueryKeys.ReadAgentInRegistry, agent],
+      enabled: !!identityItems?.length && !!registry,
+      queryFn: async () => {
+        const isAgent = await readIdentityRegistryIsAgent(
+          { args: [agent] },
+          registry,
+        );
+
+        return {
+          agent,
+          isAgent,
+        };
+      },
+      staleTime: Infinity,
+    })),
+    combine: (result) => {
```

Comment:
It's recommended to have a separate function to improve memoization, instead of anon like current one
https://tanstack.com/query/v5/docs/framework/react/reference/useQueries#memoization